backend/internal/storage/postgres/assignment_templates.go

```diff
@@ -0,0 +1,37 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/CamPlume1/khoury-classroom/internal/errs"
+	"github.com/CamPlume1/khoury-classroom/internal/models"
+)
+
+func (db *DB) AssignmentTemplateExists(ctx context.Context, templateID int64) (bool, error) {
```

Comment:
it's a little silly imo to split up a simple "create if not exists" operation into two separate DB operations